retrofit-converters/moshi/src/main/java/retrofit2/converter/moshi/MoshiResponseBodyConverter.java

```diff
@@ -39,7 +42,12 @@
       if (source.rangeEquals(0, UTF8_BOM)) {
         source.skip(UTF8_BOM.size());
       }
-      return adapter.fromJson(source);
+      JsonReader reader = JsonReader.of(source);
+      T result = adapter.fromJson(reader);
+      if (requireFullConsumption && reader.peek() == JsonReader.Token.END_DOCUMENT) {
+        throw new RuntimeException("JSON document was not fully consumed.");
```

Comment:
Hmm should this be a `JsonDataException`?

Reply:
yes.